create_bot_knowledge.py

```diff
@@ -88,126 +134,50 @@ def extract_knowledge_from_repository(project: Tuple[str, str], update_knowledge
     _LOGGER.info(f"------------------------------------------------------------------------------------")
     _LOGGER.info("Considering repo: %r" % (project[1] + "/" + project[0]))
 
-    if not update_knowledge:
-        current_path = Path().cwd()
-        source_knowledge_repo = current_path.joinpath("./Bot_Knowledge")
-
-        if not source_knowledge_repo.exists():
-            os.mkdir(source_knowledge_repo)
+    current_path = Path().cwd()
+    knowledge_dir = current_path.joinpath("./Bot_Knowledge")
+    check_directory(knowledge_dir, update_knowledge)
 
-        source_knowledge_file = source_knowledge_repo.joinpath(f'{project[1] + "-" + project[0]}.json')
-
-        if source_knowledge_file.exists():
-            _LOGGER.info(f"There is already knowledge from repo {project[1] + '/' + project[0]}")
-            _LOGGER.info(f"To update knowledge from a repo, use update_knowledge=True")
-            return 0
-
-        else:
-            _LOGGER.info(f"No previous knowledge from repo {project[1] + '/' + project[0]}")
-            _LOGGER.info(f"Starting to collect knowledge...")
-            # Retrieve list of ids of PULL REQUESTS (CLOSED)
-            pull_requests = ogr_project.get_pr_list(status=PRStatus.closed)
+    project_knowledge = knowledge_dir.joinpath(f'{project[1] + "-" + project[0]}.json')
+    
+    _LOGGER.info(f"Gathering ids of all closed PRs from {project[1] + '/' + project[0]} ...")
+    pull_requests = ogr_project.get_pr_list(status=PRStatus.closed)
 
     if update_knowledge:
-        current_path = Path().cwd()
-        source_knowledge_repo = current_path.joinpath("./Bot_Knowledge")
-
-        if not source_knowledge_repo.exists():
-            os.mkdir(source_knowledge_repo)
-        else:
-            _LOGGER.info(f"No knowledge from any repo has ever been created use update_knowledge=False first.")
-            return 0
+        with open(project_knowledge, "r") as fp:
+            data = json.load(fp)
 
-        source_knowledge_file = source_knowledge_repo.joinpath(f'{project[1] + "-" + project[0]}.json')
+        current_prs = [int(pr_id) for pr_id in data["results"].keys()]
+        _LOGGER.debug(f"Currently gathered PR ids {current_prs}")
 
-        if source_knowledge_file.exists():
-            # Retrieve list of ids of PULL REQUESTS (CLOSED)
-            pull_requests = ogr_project.get_pr_list(status=PRStatus.closed)
+        refreshed_prs = [pr.id for pr in pull_requests]
 
-            with open(source_knowledge_file, "r") as fp:
-                data = json.load(fp)
+        only_new_prs = list(set(refreshed_prs) - set(current_prs))
+        _LOGGER.debug(f"New PR ids are {only_new_prs}")
 
-            pr_id_known = [int(pr_id) for pr_id in data["results"].keys()]
-            _LOGGER.debug(f"Known PR ids {pr_id_known}")
+        #pull_requests = [pull_requests[id] for id in only_new_prs]
```

Comment:
The key to each PR should be the ID, what is that number? :)

```
{"name": "thoth-station/performance", "results": {"344627688": {"PR_labels": ["approved", "size/S"], "PR_created": 1574440983.0, "PR_approved": 1574441143.0, "PR_approved_by": "fridex", "PR_TTR": 160.0, "PR_author": "pacospace", "PR_commits_number": 1}, "344554734": {"PR_labels": ["size/XS"], "PR_created": 1574430624.0, "PR_approved": null, "PR_approved_by": null, "PR_TTR": null, "PR_author": "fridex", "PR_commits_number": 1}, "344435765": {"PR_labels": ["approved", "size/M"], "PR_created": 1574412873.0, "PR_approved": 1574429467.0, "PR_approved_by": "fridex", "PR_TTR": 16594.0, "PR_author": "pacospace", "PR_commits_number": 1}, ....}
```

Reply:
fixed, pull request number now as id